addons/binding/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/handler/SqueezeBoxServerHandler.java

```diff
@@ -572,6 +581,26 @@ private String fetchUrl(final String mac) {
             return fetchUrl(null, mac);
         }
 
+        private void handleMixerMessage(String mac, String[] messageParts) {
+            String action = messageParts[2];
+
+            switch (action) {
+                case "volume":
+                    String value = messageParts[3];
+                    updatePlayer(new PlayerUpdateEvent() {
+                        @Override
+                        public void updateListener(SqueezeBoxPlayerEventListener listener) {
+                            listener.volumeChangeEvent(mac, Integer.parseInt(value));
```

Comment:
@pgfeller  Hey Patrick, I missed this in my previous review.  But you need to catch a NumberFormatException on the call to parseInt. Otherwise, an exception will cause the listener thread to die silently.

Reply:
@mhilbush Hi Mark, thanks for the finding; I´ll include the fix in https://github.com/openhab/openhab2-addons/pull/3037

hope that is ok?

with kind regards,
Patrik

Reply:
Yeah, that's good.  

I realized last night that `mixer volume` can include incremental volume changes (+n and -n), in addition to the absolute volume value.  When I adjusted the volume from the LMS web interface, it must've sent the +2.  I guess there are two choices:
- you could filter out the +n and -n.  This is no worse than how it works today (i.e. it uses the status message to pick up the absolute volume.
- or make necessary change for the player to handle incremental volume changes, which looks like a fair bit more work.

Reply:
@mhilbush 

Hi Mark, I´ll have a look how difficult it would be to support increments and decrements as well :-). Again using the refactoring PR.